### PR TITLE
Dns discovery

### LINKED ISSUE #889: Add DNS support to discovery
## Description

Discovery is currently based on IP. The PING packets update the ENR's IP. 

A configuration should allow the Enr to have a fixed static IP, regardless of the PONG response from other peers. We should allow the static IP to be derived from a DNS name. 

When setting the `--discovery-address` flag, the Enr's IP should remain static and ignore reported IP's from PONG responses. 

We should add a `--discovery-dns-address` flag which can specify the ENR's IP from a DNS name

See https://github.com/ethereum/devp2p/issues/138

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

Resolves #889 

## Proposed Changes

Add a cli flag `discovery-dns-address` for setting a static discovery address through a dns hostname.
Setting this option will disable auto updating the discovery address based on PONG responses from connected peers in discv5. 

**Note**: Since auto-update of enr-address is disabled with this option, the `enr-udp-port` needs to be set for other nodes to be able to ping our node through discovery. 

## Additional Info

We might want to periodically resolve the dns address and update the enr ip in case of a change in a future PR.

### CODE REVIEW

**Comment by pawanjay176** [comment]
Yep, sounds good to me. Much easier than having another cli option to get confused with.
Will make the changes.

**Comment by pawanjay176** [comment]
@AgeManning  if the user inputs a static IP (not a DNS address), we don't necessarily require the `enr-udp-port` to be set right? The port can eventually resolve with auto-update. It's only when the user inputs a DNS address, that we require `enr-udp-port` to be set.

**Comment by AgeManning** [comment]
I think we do need to require a UDP port if an IP is given. If an IP is given with a UDP port, it doesn't do anything, its effectively the same as having no IP address. So the user may be confused. The address and port will get updated if enough nodes agree on our external socket regardless of if the user entered and IP or not. 

I think we should use the `.requires()` in the CI to make it clear.

File: beacon_node/src/config.rs [hunk 1]
```diff
@@ -176,6 +168,40 @@ pub fn get_config<E: EthSpec>(
...
+                // `to_socket_addr()` does the dns resolution
+                // Note: `to_socket_addrs()` is a blocking call
+                let resolved_addr = if let Ok(mut resolved_addrs) = addr.to_socket_addrs() {
+                    // Pick the first ip from the list of resolved addresses
+                    resolved_addrs
+                        .next()
+                        .map(|a| a.ip())
+                        .ok_or_else(|| format!("Resolved dns addr contains no entries"))?
+                } else {
+                    return Err("Failed to parse enr-address".into());
```
Comment (AgeManning) [inline beacon_node/src/config.rs:196]: Can we return the address that could not be passed back here so the user can see what is wrong

**Pull request merged**